.github/workflows/build_test_and_deploy.yml

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2022 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+name: Build, test and deploy docker images
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  # Build base docker image
+  build_base_image:
+    name: github/pika-ci-base/fast
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASS }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile.base
+          push: true
+          tags: ${{ secrets.DOCKER_USER }}/pika-ci-base:latest,${{ secrets.DOCKER_USER }}/pika-ci-base:1
```

Comment:
Would it be possible to have the version tag in a variable?

Reply:
yes I am looking into that, also looking for the image name

Reply:
So using the environmental variable in the deployment step is working but using it in the job name `github/pika-ci-base` makes the workflow invalid. I'll leave the name hard coded.